tests/principals/federated.spec.ts

```diff
@@ -0,0 +1,16 @@
+import {expect} from 'chai';
+import {FederatedPrincipal} from '../../src/principals/federated';
+
+describe('#ServicePrincipal', function() {
+  describe('#toJSON', function() {
+    const identityProvider = 'www.amazon.com';
+    const policy = new FederatedPrincipal(identityProvider);
+
+    it('should return the AWS principal JSON fragment', function() {
```

Comment:
I would suggest modifying the test description. I know it is the same as in `principals/service.spec.ts`. But, I realise the naming wasn't great there too.

```suggestion
it('should return the AWS federated principal JSON fragment', function() {
```